velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -684,4 +700,46 @@ RowVectorPtr NestedLoopJoinProbe::getBuildMismatchedOutput(
       pool(), outputType_, nullptr, numUnmatched, std::move(projectedChildren));
 }
 
+RowVectorPtr NestedLoopJoinProbe::getOutputLeftSemiJoinImpl() {
+  VELOX_CHECK_NOT_NULL(input_);
+
+  bool matched = false;
+  numOutputRows_ = 0;
+
+  if (isBuildSideEmpty()) {
+    matched = false;
+  } else if (joinCondition_ == nullptr) {
+    matched = true;
+  } else {
+    for (auto i = buildRow_; i < decodedFilterResult_.size(); ++i) {
+      if (isJoinConditionMatch(i)) {
+        matched = true;
+        break;
+      }
+    }
+  }
+  return makeSingleOutputRow(matched);
+}
+
+RowVectorPtr NestedLoopJoinProbe::makeSingleOutputRow(bool matched) {
+  auto matchVector = BaseVector::create(BOOLEAN(), /*size=*/1, pool());
```

Comment:
Can we respect the max number of output rows instead of producing one row at a time which is not efficient? Thanks!